management/dns_update.py

```diff
@@ -303,6 +304,32 @@ def has_rec(qname, rtype, prefix=None):
 		if not has_rec(qname, rtype):
 			records.append((qname, rtype, value, explanation))
 
+	# Adds autoconfiguration A records for all domains.
+	# mta-sts.* - required A record for mta-sts (serving the policy)
+
+	get_prim_cert = get_ssl_certificates(env)[env['PRIMARY_HOSTNAME']]
+	response = check_certificate(env['PRIMARY_HOSTNAME'], get_prim_cert['certificate'],get_prim_cert['private-key'])
+	# we don't want those records on the primary hostname 
+	# and we only want these records if the certificate is valid
+	if response[0] == 'OK':
+		mta_sts_records = [
+			("mta-sts", "A", env["PUBLIC_IP"], "Provides MTA-STS support"),
+			("mta-sts", "AAAA", env.get('PUBLIC_IPV6'), "Provides MTA-STS support"),
+			("_mta-sts", "TXT", "v=STSv1; id=%sZ" % datetime.datetime.now().strftime("%Y%m%d%H%M%S"), "Enables MTA-STS support")
+		]
+		# Skip if the user has set a custom _smtp._tls record.
+		if not has_rec("_smtp._tls", "TXT", prefix="v=TLSRPTv1;"):
+			tls_rpt_email = "tlsrpt@%s" % env['PRIMARY_HOSTNAME'] 
```

Comment:
Need a comment here explaining TLSRPT with a link to https://tools.ietf.org/html/rfc8460.

Reply:
Added the following comment.
		# Rules can be custom configured accoring to https://tools.ietf.org/html/rfc8460.
		# Skip if the rules below if the user has set a custom _smtp._tls record.
		if not has_rec("_smtp._tls", "TXT", prefix="v=TLSRPTv1;"):
			# if the alias 'tlsrpt@PRIMARY_HOSTNAME' is configured, automaticly, reporting will be enabled to this email address

Reply:
Thanks.